features/jdbc-collector/src/main/java/org/opennms/netmgt/collectd/JdbcCollector.java

```diff
@@ -241,8 +241,11 @@ public CollectionSet collect(CollectionAgent agent, EventProxy eproxy, Map<Strin
                             JdbcCollectionResource resource = null;
                             
                             // Create the appropriate resource container.
-                            if(singleInstance) {
-                                resource = nodeResource;
+                            if(null==query.getResourceType()) {
```

Comment:
By convention, we typically allow a resource type of `"node"` to refer to node level resources i.e. https://github.com/OpenNMS/opennms/blob/develop/integrations/opennms-vmware/src/main/java/org/opennms/netmgt/collectd/VmwareCollector.java#L247

We should also consider this in addition to checking for `null`.